tests/e2e-kubernetes/testsuites/credentials.go

```diff
@@ -761,6 +843,25 @@ func waitUntilRoleIsAssumable[Input any, Output any](ctx context.Context, assume
 	return output
 }
 
+// waitUntilRoleIsAssumableEKS waits until the given role is assumable.
+// This is needed because we're creating new roles in our test cases and then trying to assume those roles,
+// but there is a delay between IAM and STS services and newly created roles/policies does not appear on STS immediately.
+func waitUntilRoleIsAssumableEKS[Input any, Output any](ctx context.Context, assumeFunc func(context.Context, *Input, ...func(*eksauth.Options)) (*Output, error), input *Input) *Output {
```

Comment:
Can we re-use `waitUntilRoleIsAssumable` here by accepting error codes as input?